packages/cozy-intent/src/utils/index.ts

```diff
@@ -0,0 +1,20 @@
+export const interpolate = (
+  str: string,
+  params: Record<string, string>
+): string => {
+  if (!params) return str
+
+  const names = Object.keys(params)
+  const vals = Object.values(params)
+
+  // I do not see security risks here, we're not executing user input
+  // eslint-disable-next-line @typescript-eslint/no-implied-eval
+  const makeInterpolate = new Function(...names, `return \`${str}\`;`)
+
+  try {
+    return makeInterpolate(...vals) as string
+  } catch (error) {
+    // String template and params did not match
```

Comment:
I don't understand why we return here, instead of throwing error

Reply:
let's test this, to clarify 🙏

Reply:
It's already tested actually. It will throw when the interpolate function did not receive adequate params to create the final string. It's better to just return the template string with no variable injection than nothing (especially since it's used for logging purposes and not end-user facing)

Reply:
I'm going to expand the comment then